x/pairing/keeper/scores/score.go

```diff
@@ -0,0 +1,230 @@
+//			calculated for all providers for each slot. It depends on the slot requirements since the score of
+//			a provider can change with different requirements. For example, say that slot A requires a provider from
+//			Europe and slot B requires a provider from Asia. We have 2 possible providers: provider A which operates
+//			from Europe and and provider B that operates from Asia. Clearly, provider A's score for the Europe slot
+//			will be better than provider B's score for the Europe slot.
+//		Data structure:
+//			The PairingScore object is created for each provider, holding a pointer to the provider, its final pairing
+//			score and the score components (the provider's stake score, geo score, etc.). The ScoreComponents map keys
+//			are the requirement object's type and the map values are the score for this requirement.
+//
+// 	- Pairing score stratgy:
+//		What is it?
+// 			A Pairing score strategy (or just strategy) defines the weight of each score component in the
+// 			final score calculation for each provider. The calculation is done as follows: req1^w1 + req2^w2 + ...
+//			For example, say that the score components are stake and geolocation. To make the pairing process
+// 			biased towards stake, we could define the strategy: w_stake = 2, w_geo = 1. So the final score
+//			will be providerStakeScore^2 + providerGeoScore^1.
+//		Data structure:
+//			The ScoreStrategy object is a map in which the keys are the requirement object's type and the values
+//			are the requirement's weight.
+//
+//	Adding new requirements:
+// 	  To add a new requirement, one should create a new object that satisfies the ScoreReq interface and update the
+//	  CalcSlots() function so the new requirement will be assigned to the pairing slots (according to some logic).
+//	  Lastly, append the new requirement object to the allReqTypes var in the init() function in this file. Use
+//	  StakeReq or GeoReq as examples.
+
+package scores
+
+import (
+	"fmt"
+	"math/big"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/lavanet/lava/utils"
+	epochstoragetypes "github.com/lavanet/lava/x/epochstorage/types"
+	planstypes "github.com/lavanet/lava/x/plans/types"
+	tendermintcrypto "github.com/tendermint/tendermint/crypto"
+)
+
+var (
+	uniformStrategy ScoreStrategy
+	allReqNames     []string
+)
+
+// TODO: currently we'll use weight=1 for all reqs. In the future, we'll get it from policy
+func init() {
+	// gather all req names to a list
+	allReqNames = append(allReqNames, stake_req_name)
+
+	// init strategy
+	uniformStrategy = make(ScoreStrategy)
+	for _, reqName := range allReqNames {
+		uniformStrategy[reqName] = 1
+	}
+}
+
+// get the overall requirements from the policy and assign slots that'll fulfil them
+// TODO: this function should be changed in the future since it only supports stake reqs
+func CalcSlots(policy planstypes.Policy, minStake sdk.Int) []*PairingSlot {
```

Comment:
should `minStake` be part of the policy?
it's weird that it is passed as another arg to `CalcSlots()` - what other extra args would it take later?

Reply:
after consulting with @Yaroms, we decided to keep it as it is

Reply:
sure; please share the reasoning?

Reply:
if we're changing it, there are two options:
1. putting MinStake inside policy protobuf, which will require a migrator
2. creating a struct that will hold the policy and the minStake

even if there are other methods of solving this, Yarom didn't want to waste time on it because it looks kinda weird